### ISSUE TITLE
Messages going to Particular.ServiceControl.Errors queue

### ISSUE THREAD

**Issue by TraGicCode** [Particular/ServiceControl#1513]
How come i have messages of mine that need to be replayed showing up in the Particular.ServiceControl.Error queue and how do i replay them??

ServiceControl Version:
3.4.4

ServicePulse Version:
1.15.0

Message Content:
```
type:	<<REDACTED>>.Ordering.Contracts.Events.<<REDACTED>>.<<REDACTED>>
message_id:	7a5a1be5-0aaa-4072-ba48-a9ab01474e3d
reply_to:	APP01
correlation_id:	7a5a1be5-0aaa-4072-ba48-a9ab01474e3d
delivery_mode:	2
headers:	
NServiceBus.MessageId:	7a5a1be5-0aaa-4072-ba48-a9ab01474e3d
NServiceBus.CorrelationId:	7a5a1be5-0aaa-4072-ba48-a9ab01474e3d
NServiceBus.MessageIntent:	Publish
NServiceBus.Version:	5.2.14
NServiceBus.TimeSent:	2018-12-03 19:51:40:997919 Z
NServiceBus.ContentType:	application/json
NServiceBus.EnclosedMessageTypes:	<<REDACTED>>, <<REDACTED>>.Contracts.Events, Version=67.2018.10.29183, Culture=neutral, PublicKeyToken=null;<<REDACTED>>, <<REDACTED>>, Version=67.2018.10.29183, Culture=neutral, PublicKeyToken=null
WinIdName:	<<REDACTED>>
NServiceBus.RelatedTo:	9a1395e6-89e3-45fa-93b3-a9aa01475184
NServiceBus.ConversationId:	d78b3021-47cb-4144-b9f3-a9aa01474e09
NServiceBus.RabbitMQ.CallbackQueue:	Ordering.Host.APP01
NServiceBus.OriginatingMachine:	APP01
NServiceBus.OriginatingEndpoint:	Ordering.Host
$.diagnostics.originating.hostid:	a753ed049212624d229f8fa480a367da
NServiceBus.ReplyToAddress:	Ordering.Host.APP01
NServiceBus.NonDurableMessage:	False
NServiceBus.ProcessingStarted:	2018-12-03 19:51:40:383980 Z
NServiceBus.ProcessingEnded:	2018-12-03 19:51:40:383980 Z
$.diagnostics.hostid:	c22a13b0989a0c81625ad7feac53a1b1
$.diagnostics.hostdisplayname:	APP01
NServiceBus.ProcessingMachine:	APP01
NServiceBus.ProcessingEndpoint:	Particular.ServiceControl.prod
NServiceBus.ExceptionInfo.ExceptionType:	RabbitMQ.Client.Exceptions.AlreadyClosedException
NServiceBus.ExceptionInfo.HelpLink:	undefined
NServiceBus.ExceptionInfo.Message:	Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=0, text="End of stream", classId=0, methodId=0, cause=System.IO.EndOfStreamException: Unable to read beyond the end of the stream. 
at RabbitMQ.Client.Impl.Frame.ReadFrom(NetworkBinaryReader reader) 
at RabbitMQ.Client.Impl.SocketFrameHandler.ReadFrame() 
at RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration() 
at RabbitMQ.Client.Framing.Impl.Connection.MainLoop()
NServiceBus.ExceptionInfo.Source:	RabbitMQ.Client
NServiceBus.ExceptionInfo.StackTrace:	RabbitMQ.Client.Exceptions.AlreadyClosedException: Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=0, text="End of stream", classId=0, methodId=0, cause=System.IO.EndOfStreamException: Unable to read beyond the end of the stream. 
at RabbitMQ.Client.Impl.Frame.ReadFrom(NetworkBinaryReader reader) 
at RabbitMQ.Client.Impl.SocketFrameHandler.ReadFrame() 
at RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration() 
at RabbitMQ.Client.Framing.Impl.Connection.MainLoop() 
at RabbitMQ.Client.Framing.Impl.Connection.EnsureIsOpen() 
at RabbitMQ.Client.Framing.Impl.Connection.CreateModel() 
at NServiceBus.Transport.RabbitMQ.ConfirmsAwareChannel..ctor(IConnection connection, IRoutingTopology routingTopology, Boolean usePublisherConfirms) 
at NServiceBus.Transport.RabbitMQ.ChannelProvider.GetPublishChannel() 
at NServiceBus.Transport.RabbitMQ.MessageDispatcher.Dispatch(TransportOperations outgoingMessages, TransportTransaction transaction, ContextBag context) 
at ServiceControl.Operations.AuditIngestor.<Ingest>d__1.MoveNext() 
--- End of stack trace from previous location where exception was thrown --- 
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() 
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) 
at NServiceBus.Transport.RabbitMQ.MessagePump.<Process>d__29.MoveNext()
NServiceBus.TimeOfFailure:	2018-12-03 19:52:37:139424 Z
NServiceBus.FailedQ:	audit
NServiceBus.Transport.RabbitMQ.ConfirmationId:	1
content_type:	application/json
Payload
723 bytes
Encoding: string
{"SiteId":53,"CorrelationGuid":"77443777-858c-4221-bf58-adc891f1835f","SalesOrderNumber":"664232","CustomerEmail":"customer@yahoo.com","OrderDate":"2018-12-02T19:51:40Z","PurchaseOrders":[{"PurchaseOrderNumber":"234123-1",}
```

**Comment by TraGicCode** [comment]
Does this have anything to do with these?  Had no idea there is are 2 rabbitmq transports available.

![image](https://user-images.githubusercontent.com/6224270/49400721-c1535e00-f70a-11e8-8de1-3332c8b7bf5c.png)

**Comment by WilliamBZA** [comment]
Messages that end up in the Particular.ServiceControl.error queue are emssages that ServiceControl tried to process but couldn't for some reason.

Based on the exception message it seems that ServiceControl is unable to import audit messages, this is probably related to #1511.

**Comment by TraGicCode** [comment]
Hey @williambza,

Yes we have both error and audit forwarding enabled.  

I can double check this one I get to work but it looks like servicecontrol is forwarding this to the forwarding queue because I can see the logs of our other application picking these messages up.  I also saw what looked to be duplicate messages that my app received by servicecontrol in his forwarding queue.  

We are unaware of rabbit going down but it could be related to a direct connect that we have setup between our on premise datacenter and aws. I can look for through some logs for this.

**Comment by TraGicCode** [comment]
Hey @WilliamBZA ,

So i went and look and based on what i can tell so far 16 ( more are slowlying coming over time ) messages have shown up in the Particular.ServiceControl.errors queue.  Out of these 16 all **BUT 1** have made it to our external tool and hence are making it into the error forwarding and audit forwarding queues correctly.  Do you have any ideas on what might be causing this?

**Comment by TraGicCode** [comment]
> Messages that end up in the Particular.ServiceControl.error queue are messages that ServiceControl tried to process but couldn't for some reason.

If that's the case how did this message get here?  Did someone try and retry it but servicecontrol couldn't so servicecontrol placed it inside its error queue?

```
NServiceBus.MessageId:	28abb46e-3701-4b13-a7b3-a9ab01654483
NServiceBus.MessageIntent:	Send
NServiceBus.OriginatingSagaId:	158f13f5-dc75-468b-9051-a9ab01654475
NServiceBus.OriginatingSagaType:	CustomerRelations.Host.Sagas.Saga, CustomerRelations.Host, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
NServiceBus.ReplyToAddress:	Communications.Host
NServiceBus.CorrelationId:	50ef852c-d0ab-4ddd-97c3-a9ab010263a4
NServiceBus.OriginatingMachine:	AZ-BCP01-AP002
NServiceBus.OriginatingEndpoint:	Communications.Host
$.diagnostics.originating.hostid:	69bdad8d3302471ace722f5895d65732
NServiceBus.ContentType:	application/json
NServiceBus.EnclosedMessageTypes:	CustomerRelations.Contracts.Commands.SendEmail, CustomerRelations.Contracts, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null;CustomerRelations.Contracts.Commands.V1.ISendEmail, CustomerRelations.Contracts, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
NServiceBus.RelatedTo:	50ef852c-d0ab-4ddd-97c3-a9ab010263a4
NServiceBus.ConversationId:	5e6daeb4-a604-4583-8e44-a9ab010263a4
NServiceBus.Version:	6.4.4
NServiceBus.TimeSent:	2018-12-03 21:40:46:298551 Z
NServiceBus.NonDurableMessage:	False
NServiceBus.ProcessingStarted:	2018-12-03 21:40:47:340448 Z
NServiceBus.ProcessingEnded:	2018-12-03 21:40:50:168598 Z
$.diagnostics.hostid:	c22a13b0989a0c81625ad7feac53a1b1
$.diagnostics.hostdisplayname:	PAPP01
NServiceBus.ProcessingMachine:	PAPP01
NServiceBus.ProcessingEndpoint:	Particular.ServiceControl.real-prod
NServiceBus.ExceptionInfo.ExceptionType:	System.Exception
NServiceBus.ExceptionInfo.HelpLink:	undefined
NServiceBus.ExceptionInfo.Message:	Channel has been closed: AMQP close-reason, initiated by Peer, code=404, text="NOT_FOUND - no exchange 'Communications.Host@AZ-BCP01-AP002' in vhost 'real-prod'", classId=40, methodId=30, cause=
NServiceBus.ExceptionInfo.Source:	mscorlib
NServiceBus.ExceptionInfo.StackTrace:	System.Exception: Channel has been closed: AMQP close-reason, initiated by Peer, code=404, text="NOT_FOUND - no exchange 'Communications.Host@AP002' in vhost 'real-prod'", classId=40, methodId=30, cause= 
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() 
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) 
at ServiceControl.Operations.AuditIngestor.<Ingest>d__1.MoveNext() 
--- End of stack trace from previous location where exception was thrown --- 
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() 
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) 
at NServiceBus.Transport.RabbitMQ.MessagePump.<Process>d__29.MoveNext()
NServiceBus.TimeOfFailure:	2018-12-03 21:42:14:018134 Z
NServiceBus.FailedQ:	audit
NServiceBus.Transport.RabbitMQ.ConfirmationId:	2
content_type:	application/json
Payload
99 bytes
Encoding: string
{"SiteId":3,"OrderNumber":"7508027","CustomerEmail":"a.asdads@donuts.com","EmailType":134217728}
```

**Comment by TraGicCode** [comment]
Hey,

I don't believe duplicate messages are being sent anymore.  Please ignore that statement.

**Comment by WilliamBZA** [comment]
Just to confirm: the correct queue that these messages should be forwarded to is: `Communications.Host@AP002`, correct?

**Comment by TraGicCode** [comment]
No not at all.  Our forwarding queues are names error.forwarding and audit.forwarding.  This exchange its complaining about is related to an naervicebus host we have that sends emails which is why I’m is confusing me.  If we need to get on a call we can.  We do have a valid license of servicepulse and servicecontrol but I’m not sure the appropriate steps needed for support outside of github issues

**Comment by WilliamBZA** [comment]
Send a mail through to support@particular.net and we can dig into more detail

**Comment by TraGicCode** [comment]
Hey @boblangley ,

This can be called done once the RabbitMQ.Transport fix gets integrated into a service control release.

**Comment by danielmarbach** [comment]
Fixed as part of 3.6.3 and announced https://discuss.particular.net/t/servicecontrol-3-6-3-patch-release-available/965
